api/client/multipartwriter.go

```diff
@@ -125,6 +135,17 @@ func (m *MultipartWriter) sendJSONConfig() error {
 	return json.NewEncoder(w).Encode(m.json)
 }
 
+func (m *MultipartWriter) sendShaperAST() error {
+	if m.shaper == nil {
+		return nil
+	}
+	w, err := m.form.CreateFormField("shaper_ast")
+	if err != nil {
+		return err
+	}
+	return json.NewEncoder(w).Encode(m.shaper)
+}
+
```

Comment:
Can we drop the `sendJSONConfig` or would this be done later?

Reply:
Hmm, I'm assuming no one is using `sendJSONConfig` on multi-upload endpoint? Happy to remove right away if that's the case.

Reply:
I would assume its being used (cc: @philrz). Isn't there some default jsonConfig we use for zeek logs and suricata logs? Perhaps wait until we have a shaper that replaces the legacy config?

Reply:
It's not clear from the code fragment alone what user-facing functionality we might be talking about dropping here. If we're talking about something the Brim app uses, yes, there's built-in JSON typing config for out-of-the-box Zeek v3.x that is applied whenever a user attempts to import NDJSON (in fact, I have PR https://github.com/brimsec/brim/pull/1307 up right now for review that proposes an adjustment for it so we're better prepared for Zeek v4.x). We put that in place way back when we weren't sure how many users would bring-their-own-Zeek-TSV-logs vs. bring-their-own-Zeek-NDJSON-logs vs. bring-their-own-pcaps. By now we know that pcaps are by far what's been most popular, and at this point I can't say I'm aware of any community user that actually imported their own Zeek NDJSON logs. That doesn't mean it hasn't happened... one of the negatives of something being built-in and "just working" is that users might try it and succeed and not think twice about it. That said, knowing how brittle it is. I'd be willing to guess it's not regularly used, if ever. The one community user I _do_ know that went down the route of creating custom NDJSON type configs ended up doing it with `zq` outside of Brim to turn their NDJSON into ZNG and then importing _that_ into Brim, for example.

If it's not hurting anything to leave it alone until the ingest shaper has reached feature parity with what was there before, that might be the safest option.

Reply:
👍 agreed